### DISCUSSION TITLE
Align colors/histogram for DrawImage

### DISCUSSION THREAD

**Discussion by zimpzon** [SixLabors/ImageSharp#1278] [Q&A] [answered]
Hi, I'm using ImageSharp to blend two images and the result is what you see here. Is there a way to align the histogram/color space of the green guy to better match the background image? So he doesn't look so out of place.

![image](https://user-images.githubusercontent.com/17750350/87561331-cf29dd00-c6bc-11ea-8eeb-898330e3bce3.png)

**Comment by JimBobSquarePants** [comment, marked as answer]
We have a [`ColorMatrix`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.ColorMatrix.html) struct which you can use in conjunction with the [`Filter`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.FilterExtensions.html#methods) extension to adjust the color balance of an image.

Here's a list of preconfigured [color matrices](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.KnownFilterMatrices.html).

**Reply by zimpzon** [reply]
Part of the problem is analyzing the background to figure out which adjustments to make. Any ideas here?

**Reply by JimBobSquarePants** [reply]
You could maybe loop through the pixels of the image to determine which channel (RGB) has dominance and adjust your image to match? I.E if your background image is more blue, make the foreground image more blue.

**Reply by JimBobSquarePants** [reply]
I guess you could calculate the average luminance also and adjust your foreground image to match.

**Reply by zimpzon** [reply]
Yeah I can try that. Just wanted to know if there was something built in :)

**Reply by zimpzon** [reply]
FYI, I ended up calculating the average luminance for a crop around his position, and scaling his brightness accordingly. Looks much better than before. It was really easy with this library, so great work!

![image](https://user-images.githubusercontent.com/17750350/87640984-c8947780-c747-11ea-84d1-bd9ce866fdec.png)
![image](https://user-images.githubusercontent.com/17750350/87641015-d1854900-c747-11ea-9171-4092a543e549.png)
![image](https://user-images.githubusercontent.com/17750350/87641041-da761a80-c747-11ea-970f-708411c4d910.png)

**Reply by JimBobSquarePants** [reply]
Fantastic. I'm glad you got it figured out! 😄

**Reply by antonfirsov** [reply]
This cactus is so happy for having the correct luminance! Can't stop being amused by these pics ^^

**Reply by JimBobSquarePants** [reply]
Aye, they're brilliant aren't they! 

I thought calculating the localized luminance was such a good idea too, I would have just done the whole image.

**Comment by mrunixus** [comment]
Hi, I am new to imageSharp, I am trying to extract one of the channels information from an RGB image, and save it into a different image, the library have an specific method for that matter? Regards,

**Reply by JimBobSquarePants** [reply]
Could you please open a new discussion. Thanks!

**Comment by zimpzon** [comment]
Your friendly cactus now has a Twitter. It wanders around semi-randomly and posts a few selfies every day. It's in danish since its just a personal learning project ;)

https://twitter.com/vandrendekaktus
